src/main/java/bio/terra/cda/app/service/QueryService.java

```diff
@@ -54,24 +72,38 @@ private Job runJob(Job queryJob) {
     }
   }
 
-  public List<String> runQuery(Query query) {
-    String queryString = generateQueryClause(query);
+  private Integer defaultValue(Integer val, Integer def) {
+    if (val != null) {
+      return val;
+    } else {
+      return def;
+    }
+  }
+
+  public QueryResult runQuery(Query query, Integer offset, Integer limit) {
+
+
+    String queryString = (new QueryTranslator(this.table, query)).sql();
```

Comment:
Parens not needed here
```suggestion
    String queryString = new QueryTranslator(this.table, query).sql();
```